collect_app/src/main/java/org/odk/collect/android/widgets/utilities/FileWidgetUtils.java

```diff
@@ -1,19 +1,108 @@
 package org.odk.collect.android.widgets.utilities;
 
+import android.content.ContentValues;
+import android.content.Context;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.view.View;
+
 import androidx.annotation.NonNull;
 
+import org.javarosa.form.api.FormEntryPrompt;
 import org.odk.collect.android.utilities.FileUtil;
+import org.odk.collect.android.utilities.MediaManager;
+import org.odk.collect.android.utilities.MediaUtil;
+import org.odk.collect.android.utilities.WidgetAppearanceUtils;
+import org.odk.collect.android.views.MultiClickSafeButton;
 
 import java.io.File;
+import java.util.Locale;
+
+import timber.log.Timber;
 
 public class FileWidgetUtils {
 
     private FileWidgetUtils() {
+    }
+
+    public static void hideButtonsIfNeeded(FormEntryPrompt prompt, MultiClickSafeButton button) {
+        if (prompt.getAppearanceHint() != null
+                && prompt.getAppearanceHint().toLowerCase(Locale.ENGLISH).contains(WidgetAppearanceUtils.NEW)) {
+            button.setVisibility(View.GONE);
+        }
+    }
+
+    public static File getFile(Context context, Object object, String instanceFolder) {
+        File file = null;
+        if (object instanceof Uri) {
+            String sourcePath = getSourcePathFromUri(context, (Uri) object, MediaStore.MediaColumns.DATA);
+            String destinationPath = FileWidgetUtils.getDestinationPathFromSourcePath(sourcePath, instanceFolder);
```

Comment:
Hi @seadowg! I wanted to write tests for `getFile` in case the answer is of `Uri` type. I am using many static methods here, and I would need to mock the data returned by them. 
Is there any other way to achieve this, rather than having a wrapper method nonstatic method that calls the static method, and then mocking the non- static method?

Reply:
Maybe you don't need to fake/mock any of the static methods here? I think where static methods are fine is where they are just "helpers" for your implementation.

Reply:
Right right! I can just have the checks in the unit tests of the widgets